keycloak.js

```diff
@@ -400,10 +400,17 @@ Keycloak.prototype.loginUrl = function (uuid, redirectUrl) {
   return url;
 };
 
-Keycloak.prototype.logoutUrl = function (redirectUrl) {
-  return this.config.realmUrl +
-  '/protocol/openid-connect/logout' +
-  '?redirect_uri=' + encodeURIComponent(redirectUrl);
+Keycloak.prototype.logoutUrl = function (redirectUrl, idTokenHint) {
+  const url = new URL(this.config.realmUrl + '/protocol/openid-connect/logout');
+
+  if (idTokenHint) {
+    url.searchParams.set('id_token_hint', idTokenHint);
+    if (redirectUrl) {
+      url.searchParams.set("post_logout_redirect_uri", redirectUrl);
+    }
+  }
+
+  return url.toString();
 };
```

Comment:
Since `redirectUrl` and `idTokenHint` are now mutually exclusive we can reduce nesting a bit by checking them both in a single condition. 

Note that this is a still a breaking change of an existing public API, so this will have to be communicated to consumers of this package.

```suggestion
Keycloak.prototype.logoutUrl = function (redirectUrl, idTokenHint) {
  const url = new URL(this.config.realmUrl + '/protocol/openid-connect/logout');

  if (redirectUrl && idTokenHint) {
    url.searchParams.set('id_token_hint', idTokenHint);
    url.searchParams.set("post_logout_redirect_uri", redirectUrl);
  }

  return url.toString();
};
```

Reply:
Yes, this should work.

Reply:
@jonkoops I originally though that by remove "Defensive" you want to remove "if" entirely. Which I think should not be done. But this way is fine.